TabnineSelfHosted/src/main/java/com/tabnineSelfHosted/binary/lifecycle/UserInfoStateSingleton.kt

```diff
@@ -0,0 +1,22 @@
+package com.tabnineSelfHosted.binary.lifecycle
+
+import com.intellij.util.messages.Topic
+import com.tabnineCommon.general.TopicBasedState
+import com.tabnineSelfHosted.binary.requests.userInfo.UserInfoResponse
+import java.util.function.Consumer
+
+class UserInfoStateSingleton private constructor() :
```

Comment:
same question regarding this singleton, should it be persisted?

Reply:
In general for live data like that I think we shouldn't persist the data, and just have a loading state. Other IDEs can signout while IJ was closed, which will cause a weird startup.

Reply:
what will happen when we encounter network issues?